cleanrl/ppo_pettingzoo_ma_atari.py

```diff
@@ -219,6 +227,8 @@ def get_action_and_value(self, x, action=None):
             next_value = agent.get_value(next_obs).reshape(1, -1)
             advantages = torch.zeros_like(rewards).to(device)
             lastgaelam = 0
+            next_done = torch.maximum(next_termination, next_truncation)
```

Comment:
Could you do https://github.com/vwxyzjn/cleanrl/blob/f62ad1f942890db7b752f4ad79b65be53917d3c2/cleanrl/ppo_continuous_action.py#L213 to be consistent?

Reply:
Do you want it specifically to be numpy rather than torch as well? I'm assuming for the sake of speed it's simpler to keep things on device so I went with `torch.logical_or`, but I can change it to np if that's better for some reason

Reply:
The motivation is to *minimize the lines of code difference* between `ppo_pettingzoo_ma_atari.py` and `ppo_atari.py`. if `torch.logical_or` does not give significant end-to-end speed up, it's more important to be consistent.

Reply:
Makes sense, but there seem to be issues with most of the other options (writing it out this way so it's a bit less confusing)

1. `np.logical_or(next_termination, next truncation)`,  CI but testing on my desktop with CUDA it has an error due to different devices: `TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.`. 

2. np.logical_or(next_termination.cpu(), next_truncation.cpu() works for that line but fails later on:
`Traceback (most recent call last):
  File "/mnt/d/GitHub/PettingZoo/tutorials/CleanRL/cleanrl_advanced.py", line 267, in <module>
    rewards[t] + args.gamma * nextvalues * nextnonterminal - values[t]
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
`, where `nextnonterminal` is on CPU. 
3.  `np.logical_or()` requires the two args to be on CPU, I could convert them and then back to device, but that seems messy and inefficient: `np.logical_or(next_termination.cpu(), next_truncation.cpu()).to(device)`. 
4. `torch.logical_or(next_termination, next_truncation)` works for that line but fails when trying to do subtraction on the line below (`1.0 - next_done`), as bool tensors don't support it: ```RuntimeError: Subtraction, the `-` operator, with a bool tensor is not supported. If you are trying to invert a mask, use the `~` or `logical_not()` operator instead.```
   * Another option is to then cast it to an int array, but that doesn't seem the most concise either, especially when it's the exact same behavior as maximum.

Given this. l think my original choice `torch.maximum(next_termination, next_truncation)` makes the most sense. I also looked through the other code to find uses of `np.logical_or()` and couldn't find any. Seems bizarre to suddenly take the arrays off the CUDA device midway through the training loop anyways, but ultimately it's obviously your call.